File: C-5-SDT/OOP_and_Python_Programming/M4_Ass_1/P_Minimize_Number.py
# https://codeforces.com/group/MWSDmqGsZm/contest/219774/problem/P
sz = int(input())
num_lst = [int(x) for x in input().split()]

cnt = 0
flag = True
while flag:
    total_num_div = 0
    for i, n in enumerate(num_lst):
        # pass
        if n%2==0:
            total_num_div+=1
            num_lst[i]=n/2
        else:
            break
    if total_num_div == sz:
        cnt += 1
    else:
        flag = False
        
print(cnt)

File: CP_Sheet_PH/Loops/P_Minimize_Number.cpp
/*
________________________
|                      |
|     AB Atiq          |
|______________________|
*/

// Problem link: https://codeforces.com/group/MWSDmqGsZm/contest/219774/problem/P

#include <bits/stdc++.h>
using namespace std;

int main()
{
    int sz;
    cin >> sz;

    int arr[sz];

    for (int i = 0; i < sz; i++)
    {
        cin >> arr[i];
    }

    int cnt = 0;
    bool flag = true;

    while (flag)
    {
        int total_num_div = 0;

        for (int i = 0; i < sz; i++)
        {
            if (arr[i] % 2 == 0)
            {
                total_num_div++;
                arr[i] = arr[i] / 2;
            }
            else
            {
                flag = false;
                break;
            }
        }

        if (total_num_div == sz)
        {
            cnt++;
        }
    }

    cout << cnt << "\n";

    return 0;
}